forc-pkg/src/source/reg/mod.rs

```diff
@@ -479,6 +493,61 @@ async fn fetch(fetch_id: u64, pinned: &Pinned, ipfs_node: &IPFSNode) -> anyhow::
     Ok(path)
 }
 
+/// Fetches package from CDN as a fallback when IPFS fails
+async fn fetch_from_s3(pinned: &Pinned, path: &Path) -> anyhow::Result<()> {
+    let client = reqwest::Client::new();
```

Comment:
might want to add explicit request timeout

```suggestion
    let client = reqwest::Client::builder()
        .timeout(std::time::Duration::from_secs(180)) // 3 minute timeout
        .build()
        .context("Failed to create HTTP client")?;
```